plat/allwinner/common/sunxi_pm.c

```diff
@@ -44,16 +73,37 @@ static void __dead2 sunxi_system_reset(void)
 	panic();
 }
 
+static int sunxi_validate_ns_entrypoint(uintptr_t ns_entrypoint)
+{
+	/* The non-secure entry point must be in DRAM */
+	if (ns_entrypoint >= SUNXI_DRAM_BASE &&
+	    ns_entrypoint < SUNXI_DRAM_BASE + SUNXI_DRAM_SIZE)
```

Comment:
As written, this expression looks like it overflows and wraps around to zero. Rearranging the math avoids the problem:

```
ns_entrypoint - SUNXI_DRAM_SIZE < SUNXI_DRAM_BASE
```

Reply:
@icenowy had a much better solution, which I wanted to put on top, together with the H6 support, just using:
``if (ns_entrypoint >= SUNXI_DRAM_BASE)``
There are boards with the H6 which have 3 GB of DRAM, so the end comparison is somewhat moot. We might just add a extra check for the upper 32 bits to be zero.

Reply:
Entirely omitting the second condition (end of DRAM) should be fine, then.